src/Features/CSharp/Portable/EditAndContinue/CSharpEditAndContinueAnalyzer.cs

```diff
@@ -1139,6 +1139,36 @@ internal override bool QueryClauseLambdasTypeEquivalent(SemanticModel oldModel,
             }
         }
 
+        protected override void ReportMethodBodySyntaxRudeEditsForLambda(SyntaxNode oldLambda, SyntaxNode newLambda, Match<SyntaxNode> bodyMatch, List<RudeEditDiagnostic> diagnostics)
+        {
+            // We're only interested in local functions here
+            if (!IsLocalFunction(oldLambda))
+            {
+                return;
+            }
+
+            var bodyEditsForLambda = bodyMatch.GetTreeEdits();
+            var editMap = BuildEditMap(bodyEditsForLambda);
+            foreach (var edit in bodyEditsForLambda.Edits)
+            {
+                // We are processing edits on the method body that contains the lambda as
+                // things like local function attributes are actually edits in the containing method, not the actual
+                // lambda body.
+                // We only want to consider the edits that are related to the lambda (ie, descendants).
+                if ((edit.OldNode != null && oldLambda.Contains(edit.OldNode)) ||
+                    (edit.NewNode != null && newLambda.Contains(edit.NewNode)))
```

Comment:
Ideally this can be removed too. I think it would make sense if this method just did syntactic analysis for every method body edit, via ClassifyEdit(), but the bug mentioned above precludes this.